bin/setup_new

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env bash
+# taken from https://gist.github.com/gvarela/9370838
+
+{
+  RUBY_VERSION=$(cat .ruby-version)
+  RUBY_VERSION=${RUBY_VERSION:=2.3.0}
+  PROJECT_NAME="ifme"
+} >/dev/null 2>&1
+
+## bootstrap function called at bottom of file comment out verification steps you don't need
+init (){
+  verify_ruby_version &&
+    verify_bundled_gems
+  verify_env_file
+  copy_example_config_files
+  verify_postgres_running &&
+    verify_database_migrations
+}
+
+
+## utility functions to test dependencies
+is_darwin (){
+  PLATFORM=$(uname)
+  [ "$PLATFORM" == "Darwin" ]
+}
+
+has_homebrew (){
+  [ is_darwin ] && [ hash brew >/dev/null 2>&1 ]
+}
+
+verify_homebrew (){
+  if [ ! has_homebrew ]
+  then
+    if [ is_darwin ]
+    then
+      echo 'Please install homebrew. Install instructions at http://brew.sh/'
+    else
+      echo "Please install Ruby version $RUBY_VERSION"
+    fi
+    exit 1
+  fi
+}
+
+has_postgres (){
+  hash psql >/dev/null 2>&1
+}
+
+postgres_is_running (){
+  ps aux | grep [p]ostgres >/dev/null 2>&1
+}
+
+has_rvm (){
+  hash rvm >/dev/null 2>&1
+}
+
+update_rvm (){
+  rvm get latest
+}
+
+rvm_install_ruby_version () {
+  rvm install $RUBY_VERSION
+}
+
+has_rbenv (){
+  hash rbenv >/dev/null 2>&1
+}
+
+update_rbenv (){
+  brew update
+  brew upgrade ruby-build
+}
+
+rbenv_install_ruby_version () {
+  rbenv install $RUBY_VERSION
+  rbenv local $RUBY_VERSION
+}
+
+has_ruby_version (){
+  INSTALLED_RUBY_VERSION=$(ruby -e 'puts RUBY_VERSION')
+  [ "$INSTALLED_RUBY_VERSION" == "$RUBY_VERSION" ]
+}
+
+install_ruby_version (){
+  verify_homebrew
+
+  if [ ! has_rvm ] && [ ! has_rbenv ]
+  then
+    echo 'Please install rbenv or rvm to manage your ruby versions.'
+    exit 1
+  fi
+
+  echo "Installing Ruby Version $RUBY_VERSION"
+  has_rvm && {
+  update_rvm
+  rvm_install_ruby_version
+}
+
+has_rbenv && {
+update_rbenv
+rbenv_install_ruby_version
+  }
+}
+
+verify_ruby_version (){
+  has_ruby_version || install_ruby_version
+}
+
+verify_bundled_gems () {
+  echo 'Updating gems...'
+  {
+    (gem list -i bundler || gem install bundler) && bundle install
+  } 2>&1
+}
+
+verify_env_file (){
+  if [ ! -f .env ]
+  then
+    echo "Generating secret..."
+    SECRET=$(bundle exec rake secret)
+    echo "SECRET_KEY_BASE=$SECRET" >> .env
```

Comment:
Yeah, it looks like we're already hardcoding the secret in the `config/application.example.yml` to `"examplesecretkey123123"` so I think you're right- lets just keep that as-is and remove the secret stuff from the setup script. 

I don't think we need to break it out by environment since basically all these configuration values are dev-only anyway and I think we're using the Heroku env UI to manage production secrets.